src/main/java/dk/bec/polonez/reservationsystem/service/ReservationService.java

```diff
@@ -25,26 +28,114 @@
+    private final ModelMapper modelMapper;
+
+    //TODO: check if reservation collides with other ones on that offer during addition
+
+
+    public ReservationService(ReservationRepository reservationRepository, UserRepository userRepository, OfferRepository offerRepository, AuthService authService) {
         this.reservationRepository = reservationRepository;
         this.userRepository = userRepository;
         this.offerRepository = offerRepository;
+        this.authService = authService;
+        this.modelMapper = new ModelMapper();
+
+        PropertyMap<Reservation, ResponseReservationDto> responseReservationMap = new PropertyMap<Reservation, ResponseReservationDto>() {
+            @Override
+            protected void configure() {
+                map().setOfferId(source.getOffer().getId());
+                map().setUserId(source.getUser().getId());
+            }
+        };
+
+        modelMapper.addMappings(responseReservationMap);
+    }
+
+    public ArrayList<ResponseReservationDto> getAll(Boolean sortedByDate) throws ResponseStatusException{
+        // Only for Admin - all reservations in the system
+        if(!authService.isAdminLoggedIn())
+            throw new ResponseStatusException(HttpStatus.FORBIDDEN, "Error: You don't have a permission to see it!");
+
+        List<Reservation> reservations = reservationRepository.findAll();
+
+        if(reservations.isEmpty())
+            throw new ResponseStatusException(HttpStatus.NOT_FOUND, "Error: There are no reservations");
+
+        if(sortedByDate)
+            reservations.sort(Comparator.comparingLong(Reservation::getCreatedAt));
+
+        return reservations.stream()
+                .map(reservation -> modelMapper.map(reservation, ResponseReservationDto.class))
+                .collect(Collectors.toCollection(ArrayList::new));
+    }
+
+    //TODO: Get all my reservations DONE
+
+    public ArrayList<ResponseReservationDto> getAllMine(Boolean sortedByDate) throws ResponseStatusException {
+        User currentUser = authService.getCurrentUser();
+
+        final List<Reservation> reservations = new ArrayList<>();
+
+        if(authService.isPlaceOwnerLoggedIn())
+            offerRepository.getAllByOwner(currentUser)
+                    .forEach((reservation) -> reservations.addAll(reservationRepository.findByOffer(reservation)));
+        else
+            reservations.addAll(reservationRepository.findByUser(currentUser));
+
+        if(reservations.isEmpty())
+            throw new ResponseStatusException(HttpStatus.NOT_FOUND, "Error: There are no reservations");
+
+        if(sortedByDate)
+            reservations.sort(Comparator.comparingLong(Reservation::getCreatedAt));
+
+        return reservations.stream()
+                .map(reservation -> modelMapper.map(reservation, ResponseReservationDto.class))
+                .collect(Collectors.toCollection(ArrayList::new));
+    }
+
+
+    public ArrayList<ResponseReservationDto> getAllMineSortedByDate() {
```

Comment:
z nazwy wynika, ze to jest to samo co wyzej tylko, ze posortowane, a metoda wyzej przyjmuje parametr do tego, przekombinowane

ogolnie to te 3 metody getAll getAllMine getAllMineSortedByDate - wydaje sie, ze robia to samo, poprostu wyciagaja wszsytkie rezerwacje do ktorych uzytkownik ma dostep (sortowanie moze byc zrobione, w controllerze)